planning/behavior_path_start_planner_module/src/start_planner_module.cpp

```diff
@@ -1005,8 +1005,8 @@ std::vector<Pose> StartPlannerModule::searchPullOutStartPoseCandidates(
     }
 
     if (utils::checkCollisionBetweenFootprintAndObjects(
-          local_vehicle_footprint, *backed_pose, stop_objects_in_pull_out_lanes,
-          parameters_->collision_check_margins.back())) {
+          local_vehicle_footprint, *backed_pose, back_stop_objects_in_pull_out_lanes,
+          parameters_->back_objects_collision_check_margin)) {
```

Comment:
use this margin for all directions not only backward?

Reply:
sorry I didn't understand this part
> I think it is better to expand margin only backward and keep the other directions(forward and lateral)

But your suggestion is calculate only longitudinal length, correct?
If so, I totally agree!

Reply:
yes, not forward but only backward longitudinal margin should be `back_objects_collision_check_margin`

Reply:
these functions may be useful 
https://github.com/tier4/autoware.universe/blob/awf-latest/planning/behavior_path_planner_common/src/utils/utils.cpp#L204-L248

Reply:
I understand and totally agree with your proposal!!
And thank you for sharing useful code.

Reply:
thanks!
if it is need to check lateral collision, we have two ways. 
thease are pseudo codes
1.
```
utils::checkCollisionBetweenFootprintAndObjects(
          local_vehicle_footprint, *backed_pose, back_stop_objects_in_pull_out_lanes,
          parameters_->back_objects_collision_check_margin))

calcLongitudinalDistanceFromEgoToObject(...)
```

2.
```
backward_expanded_local_vehicle_footprint = expandPolygon(local_vehicle_footprint, ...)
utils::checkCollisionBetweenFootprintAndObjects(
          backward_expanded_local_vehicle_footprint, *backed_pose, back_stop_objects_in_pull_out_lanes,
          parameters_->back_objects_collision_check_margin))

```

Reply:
I thought the change in https://github.com/autowarefoundation/autoware.universe/pull/6149/commits/3d2df33251667536f4e2d4bd9a8887c458f09a8e is expected?

Reply:
The change skips lateral collision check but is it ok?

Reply:
sorry i missed to check following message
> if it is need to check lateral collision, we have two ways.
thease are pseudo codes

I'm assuming `back_stop_objects_in_pull_out_lanes` are located in same lane correct?
Even in this case, lateral collision should be checked with `back_objects_collision_check_margin`?

Reply:
memo:
we talked face to face.
use hard margin for all distance, and check arc length for backward